src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py

```diff
@@ -218,196 +254,187 @@ def probsolve_ivp(
 
 
-def _string2ioup(ivp, which_prior, **kwargs):
+def _string2ioup(ivp, which_prior, driftspeed):
 
-    if "driftspeed" in kwargs.keys():
-        driftspeed = kwargs["driftspeed"]
-    else:
-        driftspeed = 1.0
-    if which_prior == "ioup1":
+    if which_prior == "IOUP1":
         return pnfs.statespace.IOUP(
             1,
             ivp.dimension,
             driftspeed,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "ioup2":
+    elif which_prior == "IOUP2":
         return pnfs.statespace.IOUP(
             2,
             ivp.dimension,
             driftspeed,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "ioup3":
+    elif which_prior == "IOUP3":
         return pnfs.statespace.IOUP(
             3,
             ivp.dimension,
             driftspeed,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "ioup4":
+    else:  # which_prior == "IOUP4":
         return pnfs.statespace.IOUP(
             4,
             ivp.dimension,
             driftspeed,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    else:
-        raise RuntimeError("It should have been impossible to reach this point.")
 
 
-def _string2matern(ivp, which_prior, **kwargs):
+def _string2matern(ivp, which_prior, lengthscale):
 
-    if "lengthscale" in kwargs.keys():
-        lengthscale = kwargs["lengthscale"]
-    else:
-        lengthscale = 1.0
-    if which_prior == "matern32":
+    if which_prior == "MAT32":
         return pnfs.statespace.Matern(
             1,
             ivp.dimension,
             lengthscale,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "matern52":
+    elif which_prior == "MAT52":
         return pnfs.statespace.Matern(
             2,
             ivp.dimension,
             lengthscale,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "matern72":
+    elif which_prior == "MAT72":
         return pnfs.statespace.Matern(
             3,
             ivp.dimension,
             lengthscale,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    elif which_prior == "matern92":
+    else:  # which_prior == "MAT92":
         return pnfs.statespace.Matern(
             4,
             ivp.dimension,
             lengthscale,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
-    else:
-        raise RuntimeError("It should have been impossible to reach this point.")
 
 
-def _string2filter(_ivp, _prior, _method, **kwargs):
+def _string2filter(_ivp, _prior, _method):
```

Comment:
The separation between this function and `_create_filter` is a bit blurry, so I would suggest merging them.